scio-core/src/main/java/com/spotify/scio/transforms/BaseAsyncLookupDoFn.java

```diff
@@ -246,6 +203,7 @@ public void finishBundle(FinishBundleContext context) {
         throw new RuntimeException("Failed to process futures", e);
       } catch (ExecutionException e) {
         LOG.error("Failed to process futures", e);
+        throw new RuntimeException("Failed to process futures", e);
```

Comment:
This was probably removed due to inconsistent behaviour of the java future, throwing in `waitForFutures` if a any of the future failed.